internal/builtinfunctions/functions.go

```diff
@@ -231,21 +231,23 @@ func getStringToBoolFunction() schema.CallableFunction {
 			schema.NewStringSchema(
 				nil,
 				nil,
-				regexp.MustCompile(`[Tt]rue|TRUE|[Ff]alse|FALSE|[tTfF]|[01]$`),
+				regexp.MustCompile(`(?i)^(?:true|false|[tf]|[01])$`),
 			)},
 		schema.NewBoolSchema(),
 		true,
 		schema.NewDisplayValue(
 			schema.PointerTo("stringToBool"),
 			schema.PointerTo(
 				"Interprets the input as a boolean.\n"+
-					"Accepts `1`, `t`, `T`, `true`, `TRUE`, and `True` for true.\n"+
-					"Accepts `0`, 'f', 'F', 'false', 'FALSE', and 'False' for false.\n"+
-					"Returns an error if any other value is input.",
+					"Case insensitively accepts `\"1\"`, `\"t\"`, and `\"true\"` for `true`.\n"+
```

Comment:
"Case insensitively" is awkward. Factoring the two instances out into the introductory sentence might make this easier to read. (Examples like "True" and "False" might help, even if we avoid the sillier variants like "tRue".)

Reply:
I removed the existing parts about case insensitivity, and added a following sentence about it with an example.